### DISCUSSION TITLE
Converted CR2 Is Darker Than Thumbnail

### DISCUSSION THREAD

**Discussion by SkyeHoefling** [dlemstra/Magick.NET#994] [Help]
I have a CR2 raw image that I am trying to convert to a jpeg and the image that is being written is much darker than what is in the embedded thumbnail or outputs generated from `dcraw.exe`. I am wondering if there are additional settings that need to be toggled to match the behavior.

This image was handed to me by the QA team on the project I am working on. No one knows where it came from so it could be a bad sample set, but I am not sure how to identify that.

## dcraw.exe generated tiff
![image](https://user-images.githubusercontent.com/17751436/125544452-9782ecea-7967-450b-a5da-c72552759cb8.png)

## Magick.NET generated tiff
![image](https://user-images.githubusercontent.com/17751436/125544557-be4e8954-add6-42f4-bc20-fb1c8e78b6d3.png)

## Embedded thumbnail extracted using dcraw.exe as jpeg
![sample1 thumbnail](https://user-images.githubusercontent.com/17751436/125544464-87f40632-f684-4bab-820d-ead565f6c833.jpeg)

Here is my code
```c#
var settings = new MagickReadSettings
{
  Defines = new DngReadDefines
  {
    UseCameraWhitebalance = true,
    DisableAutoBrightness = true
  },
  Format = MagickFormat.Jpeg
};

using (var image = new MagickImage(file, settings))
{
  image.Write(outputPath);
}
```

The CR2 file is too large to upload to GitHub so I created a public link to download from OneDrive 
* [Sample1.cr2](https://hoeflingme-my.sharepoint.com/:i:/g/personal/andrew_hoefling_me/EfIVB-PcUJtNvJa6kkQxaPEBfgKZfiIZEcXUynzzcG01-Q?e=0qxgOg)

I've been researching the documentation and this GitHub and have found some useful tips such as using the `DngReadDefines` but can't seem to get this image to convert as expected.

**Comment by dlemstra** [comment]
Adding the file zipped:
[sample1.001.zip](https://github.com/dlemstra/Magick.NET/files/6963965/sample1.001.zip)
[sample1.002.zip](https://github.com/dlemstra/Magick.NET/files/6963966/sample1.002.zip)
[sample1.003.zip](https://github.com/dlemstra/Magick.NET/files/6963967/sample1.003.zip)

**Comment by dlemstra** [comment]
I don't understand what you are trying to achieve? What is the output image that you are looking for? When you don't set any of the dng read defines you will get the same output image as dcraw?

**Reply by SkyeHoefling** [reply]
How do you best simulate the behavior of dcraw with magick. I was expecting magick to generate a result that matches my first screenshot above, but it generates the 2nd screenshot above. This creates a darker image than what I was expecting. I am probably not sending the right parameters when loading the image. If I use this same file directly with LibRaw or Dcraw I get the correct image.

**Reply by dlemstra** [reply]
When I read the image without any `DngReadDefines` I get the same image as "dcraw.exe generated tiff". Have you tried that?
